src/factories/workflow-specific/PIM_WorkflowFactory.sol

```diff
@@ -0,0 +1,187 @@
+        );
+
+        // assemble fundingManager config and deploy orchestrator
+        IOrchestratorFactory_v1.ModuleConfig memory fundingManagerConfig =
+        IOrchestratorFactory_v1.ModuleConfig(
+            PIMConfig.metadata,
+            abi.encode(
+                address(issuanceToken),
+                PIMConfig.bcProperties,
+                PIMConfig.collateralToken
+            )
+        );
+        IOrchestrator_v1 orchestrator = IOrchestratorFactory_v1(
+            orchestratorFactory
+        ).createOrchestrator(
+            workflowConfig,
+            fundingManagerConfig,
+            authorizerConfig,
+            paymentProcessorConfig,
+            moduleConfigs
+        );
+
+        // get bonding curve (= funding manager) address
+        address fundingManager = address(orchestrator.fundingManager());
+        issuanceToken.setMinter(fundingManager, true);
+
+        // revoke minter role from factory
+        issuanceToken.setMinter(address(this), false);
+
+        // if renounced token flag is set, renounce ownership over token, else transfer ownership to initial admin
+        if (PIMConfig.isRenouncedIssuanceToken) {
+            _transferTokenOwnership(issuanceToken, address(0));
+        } else {
+            _transferTokenOwnership(
+                issuanceToken, PIMConfig.issuanceTokenParams.initialAdmin
+            );
+        }
+
+        // if renounced workflow flag is set, renounce admin rights over workflow, else transfer admin rights to initial admin
+        if (PIMConfig.isRenouncedWorkflow) {
+            _transferWorkflowAdminRights(orchestrator, address(0));
+        } else {
+            _transferWorkflowAdminRights(
+                orchestrator, PIMConfig.issuanceTokenParams.initialAdmin
+            );
+        }
+
+        _manageInitialCollateral(
+            fundingManager,
+            PIMConfig.collateralToken,
+            PIMConfig.bcProperties.initialCollateralSupply
+        );
+
+        emit IPIM_WorkflowFactory.PIMWorkflowCreated(
+            fundingManager,
+            address(issuanceToken),
+            msg.sender,
+            PIMConfig.recipient,
+            PIMConfig.isRenouncedIssuanceToken,
+            PIMConfig.isRenouncedWorkflow
+        );
+
+        return (orchestrator, issuanceToken);
+    }
+
+    //--------------------------------------------------------------------------
+    // onlyOwner Functions
+
+    /// @inheritdoc IPIM_WorkflowFactory
+    function setFee(uint newFee) external onlyOwner {
+        fee = newFee;
+        emit IPIM_WorkflowFactory.FeeSet(newFee);
+    }
+
+    /// @inheritdoc IPIM_WorkflowFactory
+    function withdrawFee(IERC20 token, address to) external onlyOwner {
+        token.transfer(to, token.balanceOf(address(this)));
+    }
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    function _manageInitialCollateral(
```

Comment:
Do we want or need to ensure that `initialCollateralSupply` is `>0` here? It looks to me like this could be super low or zero otherwise, which I'm not sure we want, but might be a non-technical decision as well.

Reply:
Good question, @0xNuggan could you help out here?

Reply:
@0xNuggan ping

Reply:
In my personal opinion I wouldn't impose restrictions through input validations here, because it seems to me that we are still exploring the behaviour of the curve and therefore can use the flexibility.

Reply:
supply cannot be 0, but it can be very small.

Reply:
**IF** we were to add safety configuration validation at this level, it shuld be that the initialStatic price if the chosen colateral/supply is bigger than 0

Reply:
But it's up to debate if we want to do it or enforce it through the frontend/simulation

Reply:
If initialStaticPrice < 0 breaks the underlying math, I think we should add validation. I won't reimplement/generalize some `pure` version of `getStaticPriceForBuying`, but just add an if/revert after the curve has been deployed.

Reply:
@0xNuggan Any chance that you have a config at hand that would break the invariant?

Reply:
INITIAL_SUPPLY = 62131676819845007200000;
INITIAL_RESERVE = 12413909028605033;
RESERVE_RATIO = 199_800;

-> This would give a spot price of 1. Big sells would push the spot price to 0, breaking the invariant (sell(x) => y && buy(y) => x)